src/compiler/utilitiesPublic.ts

```diff
@@ -907,7 +907,8 @@ export function getNonAssignedNameOfDeclaration(declaration: Declaration | Expre
             return nameForNamelessJSDocTypedef(declaration as JSDocEnumTag);
         case SyntaxKind.ExportAssignment: {
             const { expression } = declaration as ExportAssignment;
-            return isIdentifier(expression) ? expression : undefined;
+            const skipExpression = skipOuterExpressions(expression);
+            return isIdentifier(skipExpression) ? skipExpression : undefined;

```

Comment:
I think this isn't a breaking change? We'll be able to get the name of a declaration in *more* cases than before.